ddtrace/propagation/http.py

```diff
@@ -543,20 +550,14 @@ def my_controller(url, headers):
 
         try:
             normalized_headers = {name.lower(): v for name, v in headers.items()}
-            # Check all styles until we find the first valid match
-            # DEV: We want to check them in this specific priority order
-            if PROPAGATION_STYLE_DATADOG in config._propagation_style_extract:
-                context = _DatadogMultiHeader._extract(normalized_headers)
-                if context is not None:
-                    return context
-            if PROPAGATION_STYLE_B3 in config._propagation_style_extract:
-                context = _B3MultiHeader._extract(normalized_headers)
-                if context is not None:
-                    return context
-            if PROPAGATION_STYLE_B3_SINGLE_HEADER in config._propagation_style_extract:
-                context = _B3SingleHeader._extract(normalized_headers)
+
+            # loop through the extract propagation styles specified in order
+            for prop_style in config._propagation_style_extract:
```

Comment:
This iterates through the list of prop styles provided by the user. Do we need to verify the order?

Reply:
We should probably iterate over PROPAGATION_STYLE_ALL to ensure the prop styles are applied in the correct order

Reply:
We need to use the order that the user provided, not the order that we have those values hardcoded.

Reply:
Makes sense. I misunderstood the change